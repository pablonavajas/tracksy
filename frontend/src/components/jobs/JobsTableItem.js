import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { setCurrentJob, deleteJob } from "../../actions/jobActions";

export const JobsTableItem = ({ job, setCurrentJob, deleteJob }) => {
  return (
    <tr>
      <td>{job.title}</td>
      <td>{job.salary}</td>

      {/* Review Intros Button */}
      <td className="center">
        <a
          id="reviewIntrosButton"
          href="#review-intros-modal"
          onClick={(e) => setCurrentJob(job)}
          className="modal-trigger"
        >
          <i className="centered material-icons light-blue-text">contacts</i>
        </a>
      </td>

      {/* Delete Button */}
      <td className="center">
        <a
          id="deleteJobItemButton"
          href="#!"
          onClick={(e) => {
            e.preventDefault();
            deleteJob(job.startupId, job.id);
          }}
        >
          <i className="centered material-icons light-blue-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

JobsTableItem.propTypes = {
  job: PropTypes.object.isRequired,
  setCurrentJob: PropTypes.func.isRequired,
  deleteJob: PropTypes.func.isRequired,
};

export default connect(null, { setCurrentJob, deleteJob })(JobsTableItem);
