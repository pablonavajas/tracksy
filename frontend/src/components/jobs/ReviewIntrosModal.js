import React, { useEffect, useState, Fragment } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { updateIntroductionsStatuses } from "../../actions/introductionActions";
import { setCurrentJob } from "../../actions/jobActions";
import Preloader from "../layout/Preloader";

export const ReviewIntrosModal = ({
  jobs: { current },
  updateIntroductionsStatuses,
  setCurrentJob,
}) => {
  const [introductions, setIntroductions] = useState(null);

  useEffect(() => {
    if (current) {
      setIntroductions(current.introductions);
    }
  }, [current]);

  const setStatus = (introductionId, status) => {
    introductions.map((introduction) => {
      introduction.id === introductionId ? (introduction.status = status) : {};
    });
    setIntroductions([...introductions]);
  };

  const onSubmit = () => {
    introductions.map((introduction) =>
      updateIntroductionsStatuses(current.startupId, current.id, introduction)
    );
    setCurrentJob(null);
  };

  return (
    <div id="review-intros-modal" className="modal" style={modalStyle}>
      {!current ? (
        <Preloader />
      ) : (
        <Fragment>
          <div className="modal-content">
            <h4>Review Introductions</h4>
            <div className="col s12">
              <table className="responsive-table highlight">
                <thead>
                  <tr>
                    <th>Name</th>
                    <th>Description</th>
                    <th>Status</th>
                    <th></th>
                  </tr>
                </thead>
                <tbody>
                  {current.introductions.map((introduction) => {
                    //if the connection for this introduciton exists
                    if (introduction.connection) {
                      return (
                        <tr key={introduction.id}>
                          <td>
                            <a href={introduction.connection.url}>
                              {introduction.connection.name}
                            </a>
                          </td>
                          <td>{introduction.connection.description}</td>
                          <td>
                            <select
                              id="rateIntroductionSelection"
                              className="browser-default"
                              onChange={(e) =>
                                setStatus(introduction.id, e.target.value)
                              }
                            >
                              <option value="" disabled>
                                Choose your option
                              </option>
                              <option value="Good">Good</option>
                              <option value="Excellent">Excellent</option>
                              <option value="Bad">Bad</option>
                            </select>
                          </td>
                        </tr>
                      );
                    }
                  })}
                </tbody>
              </table>
            </div>
          </div>
          <div className="modal-footer">
            <div className="center">
              <a
                id="submitIntroductionsRatingButton"
                onClick={onSubmit}
                className="btn waves-effect waves-light light-blue modal-close"
              >
                Submit
              </a>
            </div>
          </div>
        </Fragment>
      )}
    </div>
  );
};

ReviewIntrosModal.propTypes = {
  jobs: PropTypes.object.isRequired,
  updateIntroductionsStatuses: PropTypes.func.isRequired,
  setCurrentJob: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  jobs: state.jobs,
});

const modalStyle = {
  width: "60%",
  height: "75%",
};

export default connect(mapStateToProps, {
  updateIntroductionsStatuses,
  setCurrentJob,
})(ReviewIntrosModal);
