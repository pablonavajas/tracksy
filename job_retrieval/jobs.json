[{"url": "https://angel.co/company/konghq/jobs/761395-senior-software-engineer-kubernetes-remote", "description": "Are you ready to join the API revolution?\nYou will be part of a team that builds the Kong for Kubernetes.\nYou will be part of a team that is responsible for managing and making Kong cloud-native. This team is responsible for making and pushing kong to the bleeding edge of cloud-native and the CNCF stack.\nWhat you'll be doing:\nWork on Kong for Kubernetes, Ingress controller, and helm charts\nAdd features and maintain Kong\u2019s integrations with Prometheus, Grafana, Helm, and other products in the CNCF space.\nDesign and build extend Kong for Cloud-Native Stack\nLeverage your deep experience working on large scale cloud-native infrastructure\nWhat you'll bring:\nBS degree in Computer Science, similar technical field of study or equivalent practical experience\nInterest or experience in Kubernetes, service mesh, L3/4/7 Proxy, API services, container technologies, and the Cloud Native Stack.\nExperience in running Kubernetes or having a thorough understanding of Kubernetes architecture\nSolid hands-on low level technical expertise, understanding distributed systems, networking and computer systems programming\nStrong sense of ownership and good communication skills\nBonus Points:\nA demonstrable track record of large scale C/C++/Go projects, familiar with gRPC, Containerization and Microservices\nHands on experience of building and maintain Kubernetes controllers or operators\nExperience in building and fostering an Open-Source community\nHands-on technical leadership experience\nExperience scoping, resourcing, managing and overcoming obstacles in software development projects", "title": "Senior Software Engineer - Kubernetes (Remote)", "salary": "$120k \u2013 $165k AngelList Est.", "specs": {"Location": ["United States", "Remote"], "Hires remotely": "Everywhere", "Job type": "Full-time", "Visa sponsorship": "Available"}}, {"url": "https://angel.co/company/konghq/jobs/675881-technical-solutions-engineer", "description": "Are you ready to join the API revolution?\nAbout the role:\nYou will have the opportunity to work with and support many new, leading-edge technologies that solve real customer challenges while being a voice for customers, community, partners and your team. You want to make our customers and partners successful while strengthening the relationship with Kong Inc. You can make a huge impact and have real ownership for the work you do.\nWhat you'll be doing:\nWorking with top developers from around the world; helping to implement the Kong API Gateway and Microservices solutions for their unique environments\nDiving deeply into complex technical puzzles and work collaboratively to understand and resolve customer issues\nBe the champion of our customers by providing real-time feedback to Kong about product performance and customer needs\nBe a driver of innovation within the support team, the company and with our customers, by looking for better ways to do the work we do, improve the tools we use and the how we raise the bar on the service we deliver to our Global 5000 customers\nWhat you bring:\nYou are curious learner, who loves to dig into technical mysteries and have the creativity to solve complex technical problems\nYou can build relationships with like-minded teammates, smart customers, and global partners\nSome of the skills that you will regularly use include:\nREST and RESTful APIs\nCoding skills- (at a minimum read and troubleshoot code)\nContainers, cluster deployment and management tools like Docker, Docker Swarm, Kubernetes and others\nYou know your way around web architectures including HTTP\nBonus Points:\nFamiliarity with DNS, Linux/Unix type systems and/or databases", "title": "Technical Solutions Engineer", "salary": "$105k \u2013 $160k AngelList Est.", "specs": {"Location": "San Francisco", "Job type": "Full-time", "Visa sponsorship": "Not Available"}}, {"url": "https://angel.co/company/konghq/jobs/435773-senior-cloud-engineer", "description": "Are you ready to join the API revolution?\nAbout the role:\nYou will be part of a team that builds the next generation Kong API Cloud Gateway based in San Francisco, CA.\nWhat you'll be doing:\nDesign, build and operate highly available, distributed Kong API Cloud Gateway\nCollaborate with a variety of teams to design architectures to enable them on the cloud\nBuild and deploy full stack applications worldwide\nLeverage your deep experience working on large scale cloud-native infrastructure\nWhat you bring:\n2 plus years of experience with cloud architectural patterns - HTTP/HTTPS, L3/4/7 Proxy, API services, Multi-tenancy software\nSolid hands-on low level technical expertise, understanding distributed systems, networking and computer systems programming\nStrong sense of ownership and good communication skills\nBS degree in Computer Science, similar technical field of study or equivalent practical experience\nBonus Points:\nSaaS cloud background with demonstrable track record of large scale C/C++/Go projects, familiar with gRPC, Containerization and Micro-services\nHands-on technical leadership experience\nExperience scoping, resourcing, managing and overcoming obstacles in software development projects", "title": "Senior Cloud Engineer", "salary": "$100k \u2013 $150k", "specs": {"Location": "San Francisco", "Job type": "Full-time", "Visa sponsorship": "Not Available"}}]