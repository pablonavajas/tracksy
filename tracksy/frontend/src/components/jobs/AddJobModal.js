import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { addJob, getJobDescription } from "../../actions/jobActions";
import "../smallComponents/inputFields";
import { InputField } from "../smallComponents/inputFields";

export const AddJobModal = ({
  startup: { startups },
  jobs: { latestRetrieved },
  addJob,
  getJobDescription,
}) => {
  const [url, setUrl] = useState("");
  const [description, setDescription] = useState("");
  const [title, setTitle] = useState("");
  const [salary, setSalary] = useState("");
  const [location, setLocation] = useState("");

  useEffect(() => {
    if (latestRetrieved) {
      setUrl(latestRetrieved.url);
      setDescription(latestRetrieved.description);
      setTitle(latestRetrieved.title);
      setSalary(latestRetrieved.salary);
      setLocation(latestRetrieved.specs.Location);
    }
  }, [latestRetrieved]);

  const onSubmit = () => {
    const newJob = {
      url,
      description,
      title,
      salary,
      location,
    };

    addJob(startups[0].id, newJob);

    setUrl("");
    setDescription("");
    setTitle("");
    setSalary("");
    setLocation("");
  };

  return (
    <div id="add-job-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Add Job</h4>
        <div className="row">
          <form className="col s12">
            <div className="input-field">
              {/* Button to scrape info from AngelList */}
              <a
                id="getJobDescriptionButton"
                onClick={(e) => {
                  e.preventDefault();
                  getJobDescription({
                    url,
                  });
                }}
                className="waves-effect large waves-light postfix"
              >
                <i className="material-icons center">cloud_download</i>
              </a>
              <input
                placeholder=""
                id="url"
                type="url"
                className="validate"
                value={url}
                onChange={(e) => setUrl(e.target.value)}
              />
              <label htmlFor="url">Angel List URL</label>
            </div>
            <InputField
              placeholder=""
              id="title"
              type="text"
              name="Title"
              className="input-field"
              value={title}
              setFunction={setTitle}
            />
            <InputField
              placeholder=""
              id="salary"
              type="text"
              name="Salary"
              className="input-field"
              value={salary}
              setFunction={setSalary}
            />
            <InputField
              placeholder=""
              id="location"
              type="text"
              name="Location"
              className="input-field"
              value={location}
              setFunction={setLocation}
            />
            <div className="row">
              <div className="input-field col s12">
                <textarea
                  placeholder=""
                  id="description"
                  type="text"
                  className="materialize-textarea"
                  value={description}
                  onChange={(e) => setDescription(e.target.value)}
                ></textarea>
                <label htmlFor="description">Job Description</label>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div className="modal-footer">
        <div className="center">
          <a
            id="addJobButton"
            onClick={onSubmit}
            className="btn waves-effect waves-light light-blue modal-close"
          >
            Submit
          </a>
        </div>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%",
};

AddJobModal.propTypes = {
  startup: PropTypes.object.isRequired,
  jobs: PropTypes.object.isRequired,
  addJob: PropTypes.func.isRequired,
  getJobDescription: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  startup: state.startup,
  jobs: state.jobs,
});

export default connect(mapStateToProps, { addJob, getJobDescription })(
  AddJobModal
);
