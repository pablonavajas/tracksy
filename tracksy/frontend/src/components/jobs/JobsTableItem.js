import React from "react";

function JobsTableItem() {
  return (
    <tr>
      <td>Cloud Engineer</td>
      <td>$80k</td>
      <td className="center">
        <a href="#review-intros-modal" className="modal-trigger">
          <i className="centered material-icons light-blue-text">contacts</i>
        </a>
      </td>
    </tr>
  );
}

export default JobsTableItem;
