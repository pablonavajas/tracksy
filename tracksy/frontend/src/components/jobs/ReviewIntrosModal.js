import React from "react";

function ReviewIntrosModal() {
  return (
    <div id="review-intros-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Review Introductions</h4>
      </div>
    </div>
  );
}

const modalStyle = {
  width: "60%",
  height: "75%"
};

export default ReviewIntrosModal;
