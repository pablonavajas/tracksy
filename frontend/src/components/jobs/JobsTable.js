import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import JobsTableItem from "./JobsTableItem";
import Preloader from "../layout/Preloader";

export const JobsTable = ({ startup: { startups, loading } }) => {
  if (loading) {
    return <Preloader />;
  }

  return (
    <div className="col s12">
      <table className="responsive-table highlight">
        <thead>
          <tr>
            <th>Job Title</th>
            <th>Salary</th>
            <th className="center"></th>
            <th className="center"></th>
          </tr>
        </thead>

        <tbody>
          {startups[0].jobs.map((job) => (
            <JobsTableItem job={job} key={job.id} />
          ))}
        </tbody>
      </table>
      <div className="center" style={addButtonStyle}>
        <a
          href="#add-job-modal"
          className="modal-trigger btn-floating waves-effect waves-light light-blue"
        >
          <i className="material-icons">add</i>
        </a>
      </div>
    </div>
  );
};

const addButtonStyle = {
  margin: "-1.1em ",
};

JobsTable.propTypes = {
  startup: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  startup: state.startup,
});

export default connect(mapStateToProps, null)(JobsTable);
