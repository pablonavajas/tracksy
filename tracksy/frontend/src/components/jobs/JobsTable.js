import React from "react";
import JobsTableItem from "./JobsTableItem";

function JobsTable() {
  return (
    <div className="col s12">
      <table className="responsive-table highlight">
        <thead>
          <tr>
            <th>Job Title</th>
            <th>Salary</th>
            <th className="center">Introductions</th>
          </tr>
        </thead>

        <tbody>
          <JobsTableItem />
          <tr>
            <td>Alan</td>
            <td>Jellybean</td>
            <td>$3.76</td>
          </tr>
          <tr>
            <td>Jonathan</td>
            <td>Lollipop</td>
            <td>$7.00</td>
          </tr>
        </tbody>
      </table>
      <div className="center" style={addButtonStyle}>
        <a
          href="#add-job-modal"
          className="modal-trigger btn-floating waves-effect waves-light light-blue"
        >
          <i className="material-icons">add</i>
        </a>
      </div>
    </div>
  );
}

const addButtonStyle = {
  margin: "-1.4em "
};

export default JobsTable;
