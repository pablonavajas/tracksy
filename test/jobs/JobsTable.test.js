import React from "react";
import { shallow } from "enzyme";
import { JobsTable } from "../../tracksy/frontend/src/components/jobs/JobsTable";
import "../setupTests";
import toJson from "enzyme-to-json";

//only VC Funds can register
describe("JobsTableItem", () => {
  let startup, wrapper;

  beforeEach(() => {
    startup = {
      startups: [{ id: 0, jobs: [{ id: 0, title: "testJobName" }] }],
      loading: false,
    };
  });

  it("renders correctly", () => {
    wrapper = shallow(<JobsTable startup={startup} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the preloader when loading is true", () => {
    startup.loading = true;
    wrapper = shallow(<JobsTable startup={startup} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
