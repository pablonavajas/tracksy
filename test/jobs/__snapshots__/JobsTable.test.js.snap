// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsTableItem renders correctly 1`] = `
<div
  className="col s12"
>
  <table
    className="responsive-table highlight"
  >
    <thead>
      <tr>
        <th>
          Job Title
        </th>
        <th>
          Salary
        </th>
        <th
          className="center"
        />
        <th
          className="center"
        />
      </tr>
    </thead>
    <tbody>
      <Connect(JobsTableItem)
        job={
          Object {
            "id": 0,
            "title": "testJobName",
          }
        }
        key="0"
      />
    </tbody>
  </table>
  <div
    className="center"
    style={
      Object {
        "margin": "-1.1em ",
      }
    }
  >
    <a
      className="modal-trigger btn-floating waves-effect waves-light light-blue"
      href="#add-job-modal"
    >
      <i
        className="material-icons"
      >
        add
      </i>
    </a>
  </div>
</div>
`;

exports[`JobsTableItem renders the preloader when loading is true 1`] = `<Preloader />`;
